### ISSUE TITLE
`<img role="button">` does not fail `button-name`

### ISSUE THREAD

**Issue by Malvoz** [dequelabs/axe-core#3437]
Using Lighthouse (in Chrome 100.0.4896.127):

Expectation:

I expect the following markup to fail `button-name` (with empty `alt`/without `alt`):

```html
<img role="button" tabindex="0" width="50" height="50">
```

Actual:

Does not fail `button-name`.

Motivation:

Same as `<button></button>`.

**Comment by straker** [comment]
Thanks for the issue. The reason this doesn't fail `button-name` is because images without an accessible name cannot be buttons. The [ARIA in HTML spec](https://www.w3.org/TR/html-aria) defines what roles and attributes are allowed for different elements. An [`img` with no accessible name](https://www.w3.org/TR/html-aria/#el-img-no-alt) is not allowed to have any roles. For that reason we don't treat the example markup as a button but as an image which fails the `image-alt` and `aria-allowed-role` rules.

**Comment by Malvoz** [comment]
@straker Thanks for clarifying. I'm familiar with the ARIA in HTML spec, I don't know why I didn't consider that. And this isn't the first time either... 😣
